tests/update_workflow_test.go

```diff
@@ -4684,55 +4692,42 @@ func (s *functionalSuite) TestUpdateWorkflow_StaleSpeculativeWorkflowTask_ClearM
-				Attributes: &commandpb.Command_ScheduleActivityTaskCommandAttributes{ScheduleActivityTaskCommandAttributes: &commandpb.ScheduleActivityTaskCommandAttributes{
-					ActivityId:             tv.ActivityID("3"),
-					ActivityType:           tv.ActivityType(),
-					TaskQueue:              tv.TaskQueue(),
-					ScheduleToCloseTimeout: tv.InfiniteTimeout(),
-				}},
-			}),
-		Messages:              s.acceptUpdateMessages(tv, wt4.Messages[0], "2"),
+			s.acceptUpdateCommands(tv, "1"),
+			append(
+				s.acceptUpdateCommands(tv, "2"),
+				&commandpb.Command{
+					CommandType: enumspb.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK,
+					Attributes: &commandpb.Command_ScheduleActivityTaskCommandAttributes{ScheduleActivityTaskCommandAttributes: &commandpb.ScheduleActivityTaskCommandAttributes{
+						ActivityId:             tv.ActivityID("3"),
+						ActivityType:           tv.ActivityType(),
+						TaskQueue:              tv.TaskQueue(),
+						ScheduleToCloseTimeout: tv.InfiniteTimeout(),
+					}},
+				},
+				&commandpb.Command{
+					CommandType: enumspb.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK,
+					Attributes: &commandpb.Command_ScheduleActivityTaskCommandAttributes{ScheduleActivityTaskCommandAttributes: &commandpb.ScheduleActivityTaskCommandAttributes{
+						ActivityId:             tv.ActivityID("4"),
+						ActivityType:           tv.ActivityType(),
+						TaskQueue:              tv.TaskQueue(),
+						ScheduleToCloseTimeout: tv.InfiniteTimeout(),
+					}},
+				},
+			)...),
+		Messages: append(
+			s.acceptUpdateMessages(tv, wt4.Messages[0], "1"),
+			s.acceptUpdateMessages(tv, wt4.Messages[1], "2")...),
 		ReturnNewWorkflowTask: true,
 	})
 	s.NoError(err)
 	s.NotNil(wt5Resp)
 	wt5 := wt5Resp.WorkflowTask
-	s.NotNil(wt5)
-	s.NotEmpty(wt5.TaskToken, "5th workflow task must have valid task token")
-	s.Len(wt5.Messages, 1, "5th workflow task must have a message with 2nd update")
-	s.EqualValues(15, wt5.StartedEventId)
-	s.EqualValues(14, wt5.Messages[0].GetEventId())
-	s.EqualHistory(`
-	11 WorkflowTaskCompleted
-	12 WorkflowExecutionUpdateAccepted
-	13 ActivityTaskScheduled
-	14 WorkflowTaskScheduled
-	15 WorkflowTaskStarted`, wt5.History)
-
-	// Complete WT5 should succeed.
-	_, err = s.engine.RespondWorkflowTaskCompleted(testCtx, &workflowservice.RespondWorkflowTaskCompletedRequest{
-		Namespace: s.namespace,
-		TaskToken: wt5.TaskToken,
-		Commands: append(s.acceptUpdateCommands(tv, "1"), &commandpb.Command{
-			CommandType: enumspb.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK,
-			Attributes: &commandpb.Command_ScheduleActivityTaskCommandAttributes{ScheduleActivityTaskCommandAttributes: &commandpb.ScheduleActivityTaskCommandAttributes{
-				ActivityId:             tv.ActivityID("4"),
-				ActivityType:           tv.ActivityType(),
-				TaskQueue:              tv.TaskQueue(),
-				ScheduleToCloseTimeout: tv.InfiniteTimeout(),
-			}},
-		}),
-		Messages: s.acceptUpdateMessages(tv, wt5.Messages[0], "1"),
-	})
-	s.NoError(err)
+	s.Nil(wt5)
```

Comment:
What does it mean? What could be the reason for wt5 not to be nil?

Reply:
No error?